File: common/src/main/java/com/datastrato/gravitino/dto/responses/FilesetContextResponse.java
/*
 * Copyright 2024 Datastrato Pvt Ltd.
 * This software is licensed under the Apache License version 2.
 */
package com.datastrato.gravitino.dto.responses;

import com.datastrato.gravitino.dto.file.FilesetContextDTO;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.google.common.base.Preconditions;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.ToString;

/** Response for fileset context. */
@Getter
@ToString
@EqualsAndHashCode(callSuper = true)
public class FilesetContextResponse extends BaseResponse {
  @JsonProperty("filesetContext")
  private final FilesetContextDTO context;

  /** Constructor for FilesetContextResponse. */
  public FilesetContextResponse() {
    super(0);
    this.context = null;
  }

  /**
   * Constructor for FilesetContextResponse.
   *
   * @param context the fileset context DTO.
   */
  public FilesetContextResponse(FilesetContextDTO context) {
    super(0);
    this.context = context;
  }

  /**
   * Validates the response.
   *
   * @throws IllegalArgumentException if the response is invalid.
   */
  @Override
  public void validate() throws IllegalArgumentException {
    super.validate();
    Preconditions.checkArgument(context != null, "context must not be null");
  }
}

File: clients/client-python/gravitino/dto/responses/fileset_context_response.py
"""
Copyright 2024 Datastrato Pvt Ltd.
This software is licensed under the Apache License version 2.
"""

from dataclasses import dataclass, field

from dataclasses_json import config

from gravitino.dto.fileset_context_dto import FilesetContextDTO
from gravitino.dto.responses.base_response import BaseResponse


@dataclass
class FilesetContextResponse(BaseResponse):
    """Response for fileset context."""

    _context: FilesetContextDTO = field(metadata=config(field_name="filesetContext"))

    def context(self) -> FilesetContextDTO:
        return self._context

    def validate(self):
        """Validates the response data.

        Raises:
            IllegalArgumentException if catalog identifiers are not set.
        """
        super().validate()
        assert self._context is not None, "context must not be null"
